lib/ansible/plugins/terminal/__init__.py

```diff
@@ -57,69 +58,110 @@ def __init__(self, connection):
 
     def _exec_cli_command(self, cmd, check_rc=True):
         '''
-        Executes the CLI command on the remote device and returns the output
+        Deprecated method sending commands across the connection
+
+        This method has been deprecated as of Ansible 2.5 and will be
+        removed from the code in Ansible 2.9.
+
+        Please use the cli() method instead.
 
         :arg cmd: Byte string command to be executed
         '''
         return self._connection.exec_command(cmd)
 
+    def cli(self, command, prompt=None, answer=None, send_only=False):
+        '''
+        Executes the command directly on the connection and returns the output
+
+        This is the preferred method when sending commands to the remote
+        device and supersedes the deprecated _exec_cli_command() method.
+        All terminal plugins should be updated to use this method.
+
+        :arg command: The command string to send to the device
+        :arg prompt: The regex string pattern to match a response
+        :arg answer: The answer to send if the prompt is matched
+        :arg send_only: Prevent the connection from waiting for a response
+        '''
+        kwargs = {'command': to_bytes(command), 'send_only': send_only}
+        if prompt:
+            kwargs['prompt'] = to_bytes(prompt)
+        if answer:
+            kwargs['answer'] = to_bytes(answer)
+        return to_text(self._connection.send(**kwargs))
+
     def _get_prompt(self):
-        """
+        '''
         Returns the current prompt from the device
 
         :returns: A byte string of the prompt
-        """
-        self._exec_cli_command(b'\n')
+        '''
+        self.cli(b'\n')
```

Comment:
Is the byte prefix required here, since text command will be converted to byte format with function?